### PR TITLE
Add net.imaginaryinfinity.Squiid

### PR DESCRIPTION
**Pull request by TabulateJarl8**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
    - `--share=network` is required due to inter-process communication between the components of the application
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by TabulateJarl8** [comment]
This is a reopened version of #4366 because it wouldn't allow me to add the generated sources file

**Comment by barthalion** [comment]
bot, build net.imaginaryinfinity.Squiid

**Review by hfiguiere — CHANGES_REQUESTED** [review]
I don't see any icon, desktop, or appstream file installed.

This is the build failure on x86_64.

**Comment by TabulateJarl8** [comment]
Let me know if this looks good

**Comment by hfiguiere** [comment]
why switching to PNG?

File: net.imaginaryinfinity.Squiid.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="console-application">
+  <id>net.imaginaryinfinity.Squiid</id>
+  
+  <name>Squiid</name>
+  <summary>Squiid is a modular calculator written in Rust.</summary>
```
Comment (hfiguiere) [inline net.imaginaryinfinity.Squiid.metainfo.xml:6]:
```suggestion
  <summary>Squiid is a modular calculator</summary>
```

File: net.imaginaryinfinity.Squiid.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  
+  <metadata_license>MIT</metadata_license>
+  <project_license>GPL-3.0-or-later</project_license>
+  
+  <description>
+    <p>
+      Squiid is a modular calculator written in Rust. It is currently very early in development but it is intended to be the successor to ImaginaryInfinity Calculator.
```
Comment (hfiguiere) [inline net.imaginaryinfinity.Squiid.metainfo.xml:13]:
```suggestion
      Squiid is a modular calculator. It is currently very early in development but it is intended to be the successor to ImaginaryInfinity Calculator.
```

File: net.imaginaryinfinity.Squiid.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  
+  <categories>
+    <category>Utility</category>
+    <category>Calculator</category>
+  </categories>
+  
+  <provides>
+    <binary>squiid</binary>
+  </provides>
+</component>
```
Comment (hfiguiere) [inline net.imaginaryinfinity.Squiid.metainfo.xml:27]: missing required `url`, `screenshots`, `releases`

File: net.imaginaryinfinity.Squiid.desktop
Comment (hfiguiere) [inline net.imaginaryinfinity.Squiid.desktop:1]: this file should be upstream

File: net.imaginaryinfinity.Squiid.metainfo.xml
Comment (hfiguiere) [inline net.imaginaryinfinity.Squiid.metainfo.xml:1]: this file should be upstream.

File: net.imaginaryinfinity.Squiid.desktop [hunk 4]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Version=1.0.2
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply net.imaginaryinfinity.Squiid.desktop:2]: https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s06.html

>  Version of the Desktop Entry Specification that the desktop entry conforms with. Entries that confirm with this version of the specification should use 1.5. Note that the version field is not required to be present.

**Comment by hfiguiere** [comment]
bot, build net.imaginaryinfinity.Squiid

**Comment by hfiguiere** [comment]
Seems like not all the crates are in the cargo-source

Was the Cargo.lock up to date when you generated it?

**Comment by TabulateJarl8** [comment]
It should've been up to date, I can double check it in a few hours and push the changes

File: net.imaginaryinfinity.Squiid.json [hunk 5]
```diff
@@ -36,7 +36,7 @@
         "sources": [{
                 "type": "archive",
                 "url": "https://gitlab.com/ImaginaryInfinity/squiid-calculator/squiid/-/archive/1.0.3/squiid-1.0.3.tar.gz",
-                "sha256": "c0ba6c1d2d55ef4b9dffc56a2ed887cf1c429d2c6d6a30cec5f5f75bbf587a13"
+                "sha256": "d778e19581339bca9911d6738b3db0cfb92f74259a0b7a4b659d6a5d4375030b"
```
Comment (hfiguiere) [inline net.imaginaryinfinity.Squiid.json:39]: 1. retagging like that is bad
2. it's `generated-sources.json` that needs to be updated.

**Comment by TabulateJarl8** [comment]
My bad, I completely forgot about that file

**Comment by hfiguiere** [comment]
bot, build net.imaginaryinfinity.Squiid

**Comment by TabulateJarl8** [comment]
Oh I'm really dumb, i checked for that and missed it. Sorry for all the stupid mistakes, I'll fix that in a bit

**Comment by hfiguiere** [comment]
bot, build net.imaginaryinfinity.Squiid

**Comment by TabulateJarl8** [comment]
It seems to have failed again, I have no idea why, it compiles on my machine. You it be against flathub's guidelines to build without using offline mode? I know its supposed to ensure the binary compiles, but it doesn't seem to be working in this instance

**Comment by TabulateJarl8** [comment]
Actually, the compiling seems to be an aarch64 issue. The caption text errors I can fix, but I don't know how to solce the compilation issue

**Comment by TabulateJarl8** [comment]
Is there any reason that the aarch64 build is getting the error `error: linker `aarch64-linux-gnu-gcc` not found`? Is that a configuration issue on my end?

**Comment by hfiguiere** [comment]
it should just work. not sure.

**Comment by TabulateJarl8** [comment]
Am I supposed to do anything else before this is merged? I don't know if it's just from the old requested changes, but I'm seeing PRs that were merged yesterday so I just wanted to check that I don't need to do anything else

**Pull request closed without merging**